src/plugins/vis_type_vislib/public/to_ast.ts

```diff
@@ -28,7 +33,10 @@ import { vislibVisName, VisTypeVislibExpressionFunctionDefinition } from './vis_
 import { BasicVislibParams, VislibChartType } from './types';
 import { getEsaggsFn } from './to_ast_esaggs';
 
-export const toExpressionAst: VisToExpressionAst<BasicVislibParams> = async (vis, params) => {
+export const toExpressionAst = async <TVisParams extends VisParams>(
+  vis: Vis<TVisParams>,
+  params: VisToExpressionAstParams
+) => {
```

Comment:
This seems to be more verbose than just using `VisToExpressionAst`.

Also why are you not using `BasicVislibParams` here? That just forces you to type cast unnecessarily. Where `VisParams` is essentially `Record`.

```suggestion
export const toExpressionAst: VisToExpressionAst<BasicVislibParams> = async (vis, params) => {
```

Reply:
The problem is that this `toExpressionAst` is used in different vis types - both in `area` and `goal` for instance, which has different `VisParams` (area uses `BasicVislibParams`, but goal and gauge uses `GaugeVisParams` ) - and `BasicVislibParams` is not the common part of them, they are comletely different. The same happens for `heatmap` vis type.
That's why type cast is necessary below, since `GaugeVisParams` doesn't have `seriesParams` & `valueAxes` inside.
From my point of view, ideally, different vis types should use different `toExpressionAst` to avoid inconsistencies like this one.

Reply:
Thanks for the explanation 👍🏼